src/app/application.cpp

```diff
@@ -518,53 +518,50 @@ int Application::exec(const QStringList &params)
-        msgBox.exec();
-#endif
-        return 1;
-    }
 
 #ifdef DISABLE_GUI
 #ifndef DISABLE_WEBUI
-    Preferences *const pref = Preferences::instance();
-    // Display some information to the user
-    const QString mesg = QString("\n******** %1 ********\n").arg(tr("Information"))
-        + tr("To control qBittorrent, access the Web UI at %1")
-            .arg(QString("http://localhost:") + QString::number(pref->getWebUiPort())) + '\n';
-    printf("%s", qUtf8Printable(mesg));
-
-    if (pref->getWebUIPassword() == "ARQ77eY1NUZaQsuDHbIMCA==:0WMRkYTUWVT9wVvdDtHAjU9b3b7uB8NR1Gur2hmQCvCDpm39Q+PsJRJPaCU51dEiz+dTzh8qbPsL8WkFljQYFQ==") {
-        const QString warning = tr("The Web UI administrator username is: %1").arg(pref->getWebUiUsername()) + '\n'
-            + tr("The Web UI administrator password is still the default one: %1").arg("adminadmin") + '\n'
-            + tr("This is a security risk, please consider changing your password from program preferences.") + '\n';
-        printf("%s", qUtf8Printable(warning));
-    }
+        Preferences *const pref = Preferences::instance();
+        // Display some information to the user
+        const QString mesg = QString("\n******** %1 ********\n").arg(tr("Information"))
+                             + tr("To control qBittorrent, access the Web UI at %1")
+                             .arg(QString("http://localhost:") + QString::number(pref->getWebUiPort())) + '\n';
+        printf("%s", qUtf8Printable(mesg));
+
+        if (pref->getWebUIPassword() == "ARQ77eY1NUZaQsuDHbIMCA==:0WMRkYTUWVT9wVvdDtHAjU9b3b7uB8NR1Gur2hmQCvCDpm39Q+PsJRJPaCU51dEiz+dTzh8qbPsL8WkFljQYFQ==") {
+            const QString warning = tr("The Web UI administrator username is: %1").arg(pref->getWebUiUsername()) + '\n'
+                                    + tr("The Web UI administrator password is still the default one: %1").arg("adminadmin") + '\n'
+                                    + tr("This is a security risk, please consider changing your password from program preferences.") + '\n';
+            printf("%s", qUtf8Printable(warning));
+        }
 #endif // DISABLE_WEBUI
 #else
-    m_window = new MainWindow;
+        m_window = new MainWindow;
 #endif // DISABLE_GUI
 
-    m_running = true;
+        m_running = true;
 
-    // Now UI is ready to process signals from Session
-    BitTorrent::Session::instance()->startUpTorrents();
+        // Now UI is ready to process signals from Session
+        BitTorrent::Session::instance()->startUpTorrents();
 
-    m_paramsQueue = params + m_paramsQueue;
-    if (!m_paramsQueue.isEmpty()) {
-        processParams(m_paramsQueue);
-        m_paramsQueue.clear();
+        m_paramsQueue = params + m_paramsQueue;
+        if (!m_paramsQueue.isEmpty()) {
+            processParams(m_paramsQueue);
+            m_paramsQueue.clear();
+        }
+        return BaseApplication::exec();
+    }
+    catch (const std::exception &err) {
```

Comment:
will this prevent the stackdump being generated?

Reply:
I did some tests, I put `throw std::range_error("test");` into `RequestParser::parse()` and use browser to trigger it.

* git master
  This is what I get in console:
  ```
  Exception thrown: test , receiver:  ""
  OBJECT QSocketNotifier::unnamed
    SIGNALS OUT
          <None>
    SIGNALS IN
          <None>
  ```
  and qbt continue to execute.

* This PR
  An error dialog appears and this in console:
  ```
  Qt has caught an exception thrown from an event handler. Throwing
  exceptions from an event handler is not supported in Qt.
  You must not let any exception whatsoever propagate through Qt code.
  If that is not possible, in Qt 5 you must at least reimplement
  QCoreApplication::notify() and catch all exceptions there.
  ```
  If I don't press OK on the dialog and waited a few more seconds, it prints the stackdump in console:
  ```
  terminate called after throwing an instance of 'std::range_error'
    what():  test


  *************************************************************
  Please file a bug report at http://bug.qbittorrent.org and provide the following information:

  qBittorrent version: v4.2.0alpha

  Caught signal: SIGABRT
  Stack trace:
    /usr/lib/libc.so.6 : gsignal()+0x10f  [0x7fae8499fd7f]
    (omitted)
  ```

Now it is a bit better to have qbt exit on exceptions, but 2 things still troubles me:
1. The message from Qt, I don' t fully understand it.
2. Is it possible to print the stacktrace immediately without waiting?

Reply:
Also worth noting, since now it is catching all exceptions and showing a dialog, only printing the exception message is not very helpful, we cannot know what code throws it, and user will be clueless what to do upon such message.

Reply:
I have to rethink this...
Qt is not friendly with exceptions. It's really a bad idea to throw exceptions from any code that can be executed as part of some "Qt Signal/Slot connection mechanism" driven call and not handle it in the same place.
I need to come up with some convenient way for the components to notify the app to shut down in case of fatal error (with a message to the user).

Reply:
>If that is not possible, in Qt 5 you must at least reimplement
QCoreApplication::notify() and catch all exceptions there.

Well this explains why `Application::notify()` was implemented...